deploy/awstools/afitools.py

```diff
@@ -95,18 +97,69 @@ def share_agfi_in_all_regions(agfi_id, useridlist):
         afi_id = get_afi_for_agfi(agfi_id, region)
         share_afi_with_users(afi_id, region, useridlist)
 
-def firesim_tags_to_description(buildtriplet, deploytriplet, commit):
-    """ Serialize the tags we want to set for storage in the AGFI description """
-    return """firesim-buildtriplet:{},firesim-deploytriplet:{},firesim-commit:{}""".format(buildtriplet,deploytriplet,commit)
+# max length of tag values is taken from early in FireSim when AWS tags were used and
+# had a limit of 256.  The AWS API does not document a constraint on the length of
+# description on https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_CreateFpgaImage.html
+# retrieved on 03MAR2022
+ILLEGAL_KV_CHARS=':,'
+STR256 = And(str, lambda s: len(s) < 256, lambda s: all([c not in s for c in ILLEGAL_KV_CHARS]),
+	     error=f"Must be string shorter than 256, not containing {pformat(ILLEGAL_KV_CHARS)}"
+	    )
+STRBOOL = Or(And(str, Use(distutils.util.strtobool)),bool)
+TAG_DICT_SCHEMA = Schema({
+    'firesim-buildtriplet': STR256,
+    'firesim-deploytriplet': STR256,
+    'firesim-commit': STR256,
+    'firesim-origin': STR256,
+    'firesim-ispushed': STRBOOL,
+    'top-commit': STR256,
+    'top-origin': STR256,
+    'top-ispushed': STRBOOL,
+})
```

Comment:
I like doing this because it is flexible enough to coerce the types at runtime and we're going from strings to typed dicts but perhaps https://docs.python.org/3/library/typing.html#typing.TypedDict should be used?